drivers/scheduler/k8s/k8s.go

```diff
@@ -2419,6 +2429,27 @@ func (k *K8s) createCoreObject(spec interface{}, ns *corev1.Namespace, app *spec
 		}
 		log.Infof("[%v] Created Rule: %v", app.Key, rule.GetName())
 		return rule, nil
+	} else if obj, ok := spec.(*kubevirtv1.VirtualMachine); ok {
+		// Create VirtualMachine Spec
+		if obj.Namespace != "kube-system" {
+			obj.Namespace = ns.Name
+		}
+		vm, err := k8sKubevirt.CreateVirtualMachine(obj)
+		if k8serrors.IsAlreadyExists(err) {
+			if vm, err = k8sKubevirt.GetVirtualMachine(obj.Name, obj.Namespace); err == nil {
+				log.Infof("[%v] Found existing VirtualMachine: %v", app.Key, obj.Name)
+				return vm, nil
+			}
+		}
+
+		if err != nil {
+			return nil, &scheduler.ErrFailedToScheduleApp{
+				App:   app,
+				Cause: fmt.Sprintf("Failed to create Virtualachine: %v, Err: %v", obj.Name, err),
+			}
+		}
+		log.Infof("[%v] Created VirtualMachine: %v", app.Key, obj.Name)
+		return vm, nil
```

Comment:
I was thinking of creating a separate method called `createVirtualMachineObjects` like this - https://github.com/portworx/torpedo/pull/1674/files#diff-79d9de3d00fb50e35f933299cfb3675d11d694ccb70e38f0f3fac6230e946d5bR5057-R5086

Let me know if this makes sense because Kubevirt VM is not a core k8s object and this function `createCoreObject` is mainly used for standard objects like deployment, pvc, statefulset, service, secret etc.,

Reply:
stork Rule is also in the same object which is not k8sCore.. Agree its not core, but we dont have any other kubevirt objects as of now. This can probably evolve once we integrate many kubevirt objects.. Instead of VM I think createKubevirtObject() make sense going forward.